massa-execution-exports/src/execution_info.rs

```diff
@@ -0,0 +1,343 @@
+        self.info_per_slot.insert(slot, info);
+    }
+}
+
+/// Struct to store Roll related operation
+#[derive(Debug, Clone)]
+pub enum OperationInfo {
+    /// Roll buy amount
+    RollBuy(Address, u64),
+    /// Roll sell amount
+    RollSell(Address, u64),
+}
+
+/// representation of a transfer
+#[derive(Clone, Debug, Serialize)]
+pub struct TransferHistory {
+    /// sender address
+    pub from: Option<Address>,
+    /// destination address
+    pub to: Option<Address>,
+    /// amount of mas
+    pub amount: Option<Amount>,
+    /// amount of rolls
+    pub roll_count: Option<u64>,
+    /// Transfer context
+    pub context: TransferContext,
+    /// Transfer type (mas, roll, deferred-credits)
+    pub t_type: TransferType,
+    /// Transfer id
+    pub id: Option<String>,
+    /// denunciation index
+    pub denunciation_index: Option<DenunciationIndex>,
+}
+
+impl Default for TransferHistory {
+    fn default() -> Self {
+        Self {
+            from: None,
+            to: None,
+            amount: None,
+            roll_count: None,
+            context: TransferContext::DeferredCredits,
+            t_type: TransferType::Mas,
+            id: None,
+            denunciation_index: None,
+        }
+    }
+}
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Serialize)]
+pub enum TransferType {
+    Mas,
+    Roll,
+    DeferredCredits,
+}
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Serialize)]
+pub enum TransferContext {
+    TransactionCoins(String),
+    AyncMsgCancel(Option<AsyncMessageId>, Option<String>),
+    DeferredCredits,
+    DeferredCallFail(String),
+    DeferredCallCancel(String),
+    DeferredCallCoins(String),
+    DeferredCallRegister,
+    DeferredCallStorageRefund(String),
+    OperationFee(String),
+    RollBuy(String),
+    RollSell(String),
+    RollSlash,
+    CreateSCStorage,
+    DatastoreStorage,
+    CallSCCoins(String),
+    AsyncMsgCoins(Option<AsyncMessageId>, Option<String>),
+    EndorsementCreatorReward,
+    EndorsementTargetReward,
+    BlockCreatorReward,
+    ReadOnlyBytecodeExecutionFee,
+    ReadOnlyFunctionCallFee,
+    ReadOnlyFunctionCallCoins,
+    SetBytecodeStorage,
+    AbiCallCoins,
+    AbiTransferCoins,
+    AbiTransferForCoins,
+    AbiSendMsgCoins,
+    AbiSendMsgFee,
+}
+
+/// Struct for execution info
+#[derive(Debug, Clone)]
+pub struct ExecutionInfoForSlot {
+    /// Slot
+    pub slot: Slot,
+    /// Timestamp
+    pub timestamp: MassaTime,
+    /// Reward for block producer
+    pub block_producer_reward: Option<(Address, Amount)>,
+    /// Rewards for endorsmement creators
+    pub endorsement_creator_rewards: HashMap<Address, Amount>,
+    /// Reward for endorsement target
+    pub endorsement_target_reward: Option<(Address, Amount)>,
+    /// Executed denunciation
+    pub denunciations: Vec<Result<DenunciationResult, String>>,
+    /// Executed Roll buy / sell
+    pub operations: Vec<OperationInfo>,
```

Comment:
the naming of this field is very misleading: it is called operations and yields an OperationInfo but somehow only applies to roll buy / roll sell

Reply:
renamed to roll_operations